emperor/support_files/js/color-view-controller.js

```diff
@@ -193,71 +204,154 @@ define([
    * continuous).
    */
   ColorViewController.getColorList = function(values, map, discrete, scaled) {
-    var colors = {}, numColors = values.length - 1, counter = 0;
-    var interpolator, min, max;
+    var scope = this;
+    var colors, gradientSVG;
     var scaled = scaled || false;
 
     if (_.findWhere(ColorViewController.Colormaps, {id: map}) === undefined){
       throw new Error("Could not find " + map + " as a colormap.");
     }
 
-    // 1 color and continuous coloring should return the first element of the
-    // map
-    if (numColors === 0 && discrete === false) {
+    // 1 color and continuous coloring should return the first element in map
+    if (values.length == 1 && discrete === false) {
       colors[values[0]] = chroma.brewer[map][0];
       return [colors, gradientSVG];
     }
 
-    if (discrete === false) {
+    //Call helper function to create the required colormap type
+    if (discrete) {
+      colors = ColorViewController.getDiscreteColors(values, map);
+    } else if (scaled) {
+      var info = ColorViewController.getScaledColors(values, map);
+      colors = info[0];
+      gradientSVG = info[1];
+    } else {
+      colors = ColorViewController.getInterpolatedColors(values, map)
+    }
+    return [colors, gradientSVG];
+  };
+
+  /**
+   *
+   * Retrieve a discrete color.
+   *
+   * @param {values} list of objects to generate a color for, usually a
+   * category in a given metadata column.
+   * @param {map} string, name of the discrete color map to use.
+   *
+   * This function will generate a list of coloring values depending on the
```

Comment:
Just noticed this block is out of date, the map is an argument and does not depend on the "system".